File: backend/mqtt_config.py
# MQTT Configuration
# Switch between public broker (internet) and local broker (private network)

# ===== DEPLOYMENT MODE =====
# Options: "public" or "local"
DEPLOYMENT_MODE = "public"

# ===== PUBLIC BROKER (Internet) =====
# Use this for distributed deployment over internet
# Both edge and dashboard connect to public broker
PUBLIC_MQTT_BROKER = "broker.hivemq.com"
PUBLIC_MQTT_PORT = 8000  # WebSocket port for frontend
PUBLIC_MQTT_PORT_TCP = 1883  # TCP port for backend

# ===== LOCAL BROKER (Private Network) =====
# Use this when running Mosquitto on edge device
# Dashboard connects to edge device's IP
# Find edge IP with: ipconfig (Windows) or ifconfig (Mac/Linux)
LOCAL_MQTT_BROKER = "192.168.1.100"  # ← CHANGE THIS to your edge device IP
LOCAL_MQTT_PORT = 8000  # WebSocket port for frontend
LOCAL_MQTT_PORT_TCP = 1883  # TCP port for backend

# ===== AUTO-SELECT BASED ON MODE =====
if DEPLOYMENT_MODE == "local":
    MQTT_BROKER = LOCAL_MQTT_BROKER
    MQTT_PORT = LOCAL_MQTT_PORT_TCP
    MQTT_WS_PORT = LOCAL_MQTT_PORT
    print(f"🏠 LOCAL MODE: Connecting to {MQTT_BROKER}:{MQTT_PORT}")
else:
    MQTT_BROKER = PUBLIC_MQTT_BROKER
    MQTT_PORT = PUBLIC_MQTT_PORT_TCP
    MQTT_WS_PORT = PUBLIC_MQTT_PORT
    print(f"🌐 PUBLIC MODE: Connecting to {MQTT_BROKER}:{MQTT_PORT}")

# ===== TOPICS (Same for both modes) =====
TOPIC_DATA = "iot/pdm/project/data"
TOPIC_RESULT = "iot/pdm/project/result"
TOPIC_EXPLAIN_REQ = "iot/pdm/project/explain"
TOPIC_EXPLAIN_RES = "iot/pdm/project/explanation"
TOPIC_RETRAIN_REQ = "iot/pdm/project/retrain"
TOPIC_RETRAIN_RES = "iot/pdm/project/retrain_result"
TOPIC_FEEDBACK = "iot/pdm/project/feedback"

File: frontend/src/mqttConfig.js
// MQTT Broker Configuration for Frontend
// Switch between public broker (internet) and local broker (private network)

// ===== DEPLOYMENT MODE =====
// Options: "public" or "local"
const DEPLOYMENT_MODE = "public"

// ===== PUBLIC BROKER (Internet) =====
// Use this for distributed deployment over internet
const PUBLIC_BROKER = {
  host: "broker.hivemq.com",
  port: 8000,
  url: "ws://broker.hivemq.com:8000/mqtt"
}

// ===== LOCAL BROKER (Private Network) =====
// Use this when running Mosquitto on edge device
// !! CHANGE THE IP ADDRESS to your edge device's IP !!
// Find edge IP with: ipconfig (Windows) or ifconfig (Mac/Linux)
const LOCAL_BROKER = {
  host: "192.168.1.100",  // ← CHANGE THIS to your edge device IP
  port: 8000,
  url: "ws://192.168.1.100:8000/mqtt"
}

// ===== AUTO-SELECT BASED ON MODE =====
export const BROKER_CONFIG = DEPLOYMENT_MODE === "local" ? LOCAL_BROKER : PUBLIC_BROKER

// Display which mode is active
console.log(`🔧 MQTT Config: ${DEPLOYMENT_MODE.toUpperCase()} mode`)
console.log(`📡 Connecting to: ${BROKER_CONFIG.url}`)

// ===== TOPICS (Same for both modes) =====
export const TOPICS = {
  DATA: "iot/pdm/project/data",
  RESULT: "iot/pdm/project/result",
  EXPLAIN_REQ: "iot/pdm/project/explain",
  EXPLAIN_RES: "iot/pdm/project/explanation",
  RETRAIN_REQ: "iot/pdm/project/retrain",
  RETRAIN_RES: "iot/pdm/project/retrain_result",
  FEEDBACK: "iot/pdm/project/feedback"
}
